custom_components/google_home/api.py

```diff
@@ -397,14 +328,35 @@ async def post(
         resp = None
 
         try:
-            async with self._session.post(
-                url, json=data, headers=HEADERS, timeout=TIMEOUT
+            async with self._session.request(
+                method, url, json=data, headers=headers, timeout=TIMEOUT
             ) as response:
                 if response.status == HTTP_OK:
                     try:
                         resp = await response.json()
                     except ContentTypeError:
-                        resp = True
+                        resp = {}
+                    device.available = True
+                elif response.status == HTTP_UNAUTHORIZED:
+                    # If token is invalid - force reload homegraph providing new token
+                    # and rerun the task.
+                    if polling:
+                        _LOGGER.debug(
+                            (
+                                "Failed to fetch data from %s due to invalid token. "
+                                "Will refresh the token and try again."
+                            ),
+                            device.name,
+                        )
+                    else:
+                        _LOGGER.warning(
+                            "Failed to send the request to %s due to invalid token. "
+                            "Please try again.",
```

Comment:
Maybe something more explicit that the integration will refresh tokens
```suggestion
                            "We will refresh the token, please try again later.",
```